python/lsst/cp/pipe/utils.py

```diff
@@ -1367,3 +1373,97 @@ def getReadNoise(exposure, ampName, taskMetadata=None, log=None):
     log.warning("Median readout noise from ISR metadata for amp %s "
                 "could not be found.", ampName)
     return np.nan
+
+
+def ampOffsetGainRatioFixup(ptc, minAdu, maxAdu, log=None):
+    """Apply gain ratio fixup using amp offsets.
+
+    Parameters
+    ----------
+    ptc : `lsst.ip.isr.PhotonTransferCurveDataset`
+        Input PTC. Will be modified in place.
+    minAdu : `float`
+        Minimum number of ADU in mean of amplifier to use for
+        computing gain ratio fixup.
+    maxAdu : `float`
+        Maximum number of ADU in mean of amplifier to use for
+        computing gain ratio fixup.
+    log : `lsst.utils.logging.LsstLogAdapter`, optional
+        Log object.
+    """
+    # We need to find the reference amplifier.
+    # Find an amp near the middle to use as a pivot.
+    if log is None:
+        log = logging.getLogger(__name__)
+
+    gainArray = np.zeros(len(ptc.ampNames))
+    for i, ampName in enumerate(ptc.ampNames):
+        gainArray[i] = ptc.gain[ampName]
+    good, = np.where(np.isfinite(gainArray))
```

Comment:
Is this the best way to define the pivot amp? Pretty much every amp will have a finite gain, even if it's bad. Perhaps do some kind of gain outlier rejection in case there is a really weird amp?

Reply:
I've added more comments around here that this is just finding the "good" (not NaN) amps and below we choose the pivot as the one that is closest to the median, which is a simple case of outlier rejection.  I also updated the test to make one of the amps way out of family and it still does work.